src/thunderbots/software/test/ai/world/robot.cpp

```diff
@@ -401,3 +401,77 @@ TEST_F(RobotTest, equality_operator_robots_with_different_timestamp)
+    EXPECT_EQ(prevPositions, robot.getPreviousPositions());
+}
+
+TEST_F(RobotTest, get_velocity_history)
+{
+    std::vector prevVelocities = {Vector(-3, 1), Vector(2.2, -0.05), Vector(2.3, -0.05)};
+
+    Robot robot = Robot(0, Point(3, 1.2), Vector(-3, 1), Angle::ofDegrees(0),
+                        AngularVelocity::ofDegrees(25), current_time, 3);
+    robot.updateState(Point(-1.2, 3), Vector(2.2, -0.05), Angle::quarter(),
+                      AngularVelocity::ofRadians(1.1), half_second_future);
+    robot.updateState(Point(-1.3, 3), Vector(2.3, -0.05), Angle::quarter(),
+                      AngularVelocity::ofRadians(1.1), half_second_future);
+
+    EXPECT_EQ(prevVelocities, robot.getPreviousVelocities());
+}
+
+TEST_F(RobotTest, get_orientation_history)
+{
+    std::vector prevOrientations = {Angle::ofDegrees(0), Angle::quarter(),
+                                    Angle::quarter()};
+
+    Robot robot = Robot(0, Point(3, 1.2), Vector(-3, 1), Angle::ofDegrees(0),
+                        AngularVelocity::ofDegrees(25), current_time, 3);
+    robot.updateState(Point(-1.2, 3), Vector(2.2, -0.05), Angle::quarter(),
+                      AngularVelocity::ofRadians(1.1), half_second_future);
+    robot.updateState(Point(-1.3, 3), Vector(2.3, -0.05), Angle::quarter(),
+                      AngularVelocity::ofRadians(1.1), half_second_future);
+
+    EXPECT_EQ(prevOrientations, robot.getPreviousOrientations());
+}
+
+TEST_F(RobotTest, get_angular_velocity_history)
+{
+    std::vector prevAngularVelocities = {AngularVelocity::ofDegrees(25),
+                                         AngularVelocity::ofRadians(1.1),
+                                         AngularVelocity::ofRadians(1.2)};
+
+    Robot robot = Robot(0, Point(3, 1.2), Vector(-3, 1), Angle::ofDegrees(0),
+                        AngularVelocity::ofDegrees(25), current_time, 3);
+    robot.updateState(Point(-1.2, 3), Vector(2.2, -0.05), Angle::quarter(),
+                      AngularVelocity::ofRadians(1.1), half_second_future);
+    robot.updateState(Point(-1.3, 3), Vector(2.3, -0.05), Angle::quarter(),
+                      AngularVelocity::ofRadians(1.2), half_second_future);
+
+    EXPECT_EQ(prevAngularVelocities, robot.getPreviousAngularVelocities());
+}
+
+TEST_F(RobotTest, get_timestamp_history)
+{
+    std::vector prevAngularVelocities = {current_time, half_second_future,
+                                         half_second_future};
+
+    Robot robot = Robot(0, Point(3, 1.2), Vector(-3, 1), Angle::ofDegrees(0),
+                        AngularVelocity::ofDegrees(25), current_time, 3);
+    robot.updateState(Point(-1.2, 3), Vector(2.2, -0.05), Angle::quarter(),
+                      AngularVelocity::ofRadians(1.1), half_second_future);
+    robot.updateState(Point(-1.3, 3), Vector(2.3, -0.05), Angle::quarter(),
+                      AngularVelocity::ofRadians(1.2), half_second_future);
+
+    EXPECT_EQ(prevAngularVelocities, robot.getPreviousTimestamps());
+}
```

Comment:
Optional; but might want to add another test for `updateState` to make sure we're properly appending histories.

Reply:
Doesn't the existing tests at the top of this file cover that? They construct a robot then call updateState().

Reply:
Ah, you're totally correct, sorry about that.